topology/probes/k8s/probe.go

```diff
@@ -95,3 +101,38 @@ func NewProbe(g *graph.Graph, manager string, subprobes map[string]Subprobe, lin
 		linkers:   linkers,
 	}
 }
+
+func logOnError(err error) {
+	logging.GetLogger().Warning(err)
+}
+
+type errorThrottle struct {
+	period   time.Duration
+	lastLock sync.Mutex
+	last     time.Time
+}
+
+func (r *errorThrottle) onError(error) {
+	r.lastLock.Lock()
+	defer r.lastLock.Unlock()
+	d := time.Since(r.last)
+	if d < r.period {
+		time.Sleep(r.period - d)
```

Comment:
time.Sleep() must not be called under lock
even if you got the model from k8s, doesn't mean it's good one

Reply:
To throttling the log output as the k8s client is event based only and you can't be hooked / do the right behavior due to k8s client design (need to be patched)

Please implement the throttling this way

```
func (r *errorThrottle) onError(error) {
	r.lastLock.RLock()
	d := time.Since(r.last)
        r.lastLock.RUnlock()

	if d < r.period {
		time.Sleep(r.period - d)
	}

	r.lastLock.Lock()
	r.last = time.Now()
	r.lastLock.Unlock()
}
```

Reply:
ok - fixed